source/math/FloatMatrix.ooc

```diff
@@ -45,6 +45,29 @@ FloatMatrix : cover {
 		result
 	}
 
+	setVertical: func (xOffset, yOffset: Int, vector: FloatPoint3D) {
+		version (safe) {
+			if (xOffset < 0 || xOffset >= this width)
+				raise("Column index out of range in FloatMatrix setVertical")
+		}
+		if (this height - yOffset < 3)
+			raise("Element positions exceed matrix dimensions in FloatMatrix setVertical")
```

Comment:
Well, it's not entirely clear to me either, but, the rationale for the `safe` block is performance - an `if` clause that needs to be run a lot will make the program run slower, and in most cases we should be able to index things correctly without having to check indices while running the program. `safe` blocks should only be active during development. Does that answer your question?